src/textformatter.ts

```diff
@@ -1,28 +1,12 @@
 // [VexFlow](http://vexflow.com) - Copyright (c) Mohit Muthanna 2010.
 //
-// This file handles a registry of text font metric information, so all
-// VexFlow modules can take advantage of font metrics in a uniform way.
-//
 
 import { RuntimeError, log } from './util';
-import { Font, FontInfo, FontGlyph, FontStyle } from './font';
-import { PetalumaScriptTextMetrics } from './fonts/petalumascript_textmetrics';
-import { RobotoSlabTextMetrics } from './fonts/robotoslab_textmetrics';
-
-/*
-// RONYEH-FONT: Use FontGlyph instead.
-export interface TextFontMetrics {
-  x_min: number;
-  x_max: number;
-  y_min: number;
-  y_max: number;
-  ha: number;
-  leftSideBearing: number;
-  advanceWidth: number;
-}
-*/
+import { Font, FontInfo, FontGlyph } from './font';
+import { PetalumaScriptFont } from './fonts/petalumascript_glyphs';
+import { RobotoSlabFont } from './fonts/robotoslab_glyphs';
 
-export interface TextFontInfo extends Record<string, unknown> {
+export interface TextFormatterInfo extends Record<string, unknown> {
   name?: string;
   resolution?: number;
   glyphs?: Record<string, FontGlyph>;
```

Comment:
Question:  Is there ever a test case where a glyph from a text font is rendered or used?  I was thinking you'd just put the characters in a 'text' element, and use the metrics to format it.

Reply:
> I was thinking you'd just put the characters in a 'text' element, and use the metrics to format it.

**This actually reminds me...!**

When do we actually need to use metrics to format text? Isn't it the browser's job to handle text rendering for us? If we ensure that a FontFace is loaded and ready to use, shouldn't Canvas fillText() and SVG \<text\> be sufficient to render text properly?

I guess for ChordSymbols, the vertical stacking probably needs information about the height of various text blocks. And  drawing the closing parentheses around the ( ♭9 / ♯11 ) probably needs the max width of the glyph + text.

Reply:
Thanks, it makes sense, I just didn't recognize it in typescript form.

>> When do we actually need to use metrics to format text?

Text is a modifier, so it goes through the same formatting pipeline as other modifiers like accidentals.  We need to calculate the bounding box and left extent of the text to place it properly.  For modifiers like annotations, the width of the text is considered when calculating the with of the tick context.

Without text metrics, you have 2 choices.  Either treat all text like the largest character (e.g. 'H'), so the note attached to an annotation would end up taking up way too much space.  This was the case for annotations (which I use for lyrics).

The other option is to render the text, measure the bounding box, and then format.  This is also problematic: it is very slow, the font may still be loading at render time, and you may not yet have an SVG/canvas on which to render.